src/Stan/Config.hs

```diff
@@ -0,0 +1,55 @@
+{- |
+Copyright: (c) 2020 Kowainik
+SPDX-License-Identifier: MPL-2.0
+Maintainer: Kowainik <xrom.xkov@gmail.com>
+
+@stan@ configurations.
+-}
+
+module Stan.Config
+    ( -- * Data types
+      Config (..)
+    , Check (..)
+    , CheckType (..)
+    , CheckFilter (..)
+    , CheckScope (..)
+    ) where
+
+import Stan.Category (Category (..))
+import Stan.Core.Id (Id (..))
+import Stan.Core.ModuleName (ModuleName (..))
+import Stan.Inspection (Inspection (..))
+import Stan.Observation (Observation (..))
+import Stan.Severity (Severity (..))
+
+
+data Config = Config
+    { configChecks :: ![Check]
+    -- , configGroupBy :: !GroupBy
+    } deriving stock (Show)
+
+data CheckType
+    = Include
+    | Ignore
+    deriving stock (Show, Eq, Enum, Bounded)
+
+-- TODO: it doesn't make sense to have both Nothing, so we need a data type like 'These'
+-- data These a b = This a | That b | Both a b
```

Comment:
I think this `TODO` can be removed. I can imagine the following use-case: users comment out some parts of existing checks for some temporary experiments. In this case, checks with both `Nothing` just won't affect anything. What do you think?